src/core/algorithms/gwo.h

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright (c) 2018, Jairo Rojas-Delgado <jrdelgado@uci.cu>
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+* Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+* Neither the name of the <organization> nor the
+names of its contributors may be used to endorse or promote products
+derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#ifndef DNN_OPT_CORE_ALGORITHMS_GWO
+#define DNN_OPT_CORE_ALGORITHMS_GWO
+
+#include <core/base/algorithm.h>
+#include <core/generators/uniform.h>
+#include <core/generators/constant.h>
+
+namespace dnn_opt
+{
+namespace core
+{
+namespace algorithms
+{
+
+/**
+ * @brief The GWO class implements an optimization metaheuristic algorithm
+ * called Particle Swarm Optimization (GWO). This is a population based on
+ * collective intelligence, algorithm inspired in the hunting behavior of
+ * gray wolves.
+ * @author Randy Alonso Benitez <rbenitez@estudiantes.uci.cu>
```

Comment:
Please leave a single empty line before the author clause. Also there are a few missing items here like date and version number. Please check code standards.